AirCasting/Services/UpdateSessionParams.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  UpdateSessionParams.swift
+//  AirCasting
+//
+//  Created by Lunar on 01/04/2021.
+//
+
+import Foundation
+import CoreData
+
+class UpdateSessionParamsService {
+    
+    var context: NSManagedObjectContext {
+        PersistenceController.shared.container.viewContext
+    }
+    
+    func updateSessionsParams(session: Session, output: FixedSession.FixedMeasurementOutput) {
+        let dateFormatter = ISO8601DateFormatter.defaultLong
+    
+        session.id = Int64(output.id)
+        session.uuid = output.uuid
+        session.type = SessionType.from(string: output.type)?.rawValue ?? -1
+    
+        session.name = output.title
+        session.tags  = output.tag_list
+        session.startTime  = dateFormatter.date(from: output.start_time)
+        session.endTime  = dateFormatter.date(from: output.end_time)
+        session.version = Int16(output.version)
+    
+        for (_, streamOutput) in output.streams {
+            let stream: MeasurementStream = self.context.newOrExisting(id: streamOutput.id)
+            
+            stream.sensorName = streamOutput.sensor_name
```

Comment:
Have you considered checking if the stream exists in the db and if it does then just saving the measurement and not update or of the other information? If that would work then it would help with the issue of resetting thresholds set by the user